packages/eas-cli/src/commands/update/index.ts

```diff
@@ -651,6 +692,99 @@ export default class UpdatePublish extends EasCommand {
       nonInteractive,
       emitMetadata,
       json: flags.json ?? false,
+      withEasEnvironmentVariablesSet: flags[
+        'with-eas-environment-variables-set'
+      ] as EnvironmentVariableEnvironment,
+    };
+  }
+}
+
+function isEnvironmentVariableEnvironment(env: string): env is EnvironmentVariableEnvironment {
+  return Object.values(EnvironmentVariableEnvironment).includes(
+    env as EnvironmentVariableEnvironment
+  );
+}
+
+async function maybeUpdateProcessEnvWithEasEnvironmentVariablesAsync({
+  projectId,
+  environment,
+  graphqlClient,
+}: {
+  projectId: string;
+  environment: EnvironmentVariableEnvironment;
+  graphqlClient: ExpoGraphqlClient;
+}): Promise<void> {
+  if (environment) {
+    Log.log(
+      `Using server-side EAS environment variables for ${environment} environment. When using EAS environment variables, eas update command won't use env variables from .env files when creating update bundles.`
+    );
+    const environmentVariables = await EnvironmentVariablesQuery.byAppIdWithSensitiveAsync(
+      graphqlClient,
+      {
+        appId: projectId,
+        environment,
+      }
+    );
+    const readableEnvironmentVariables = Object.fromEntries(
+      environmentVariables
+        .filter(({ value }) => !!value)
+        .map(({ name, value }) => [name, value?.toString()])
+    );
+    const secretEnvironmentVariables = Object.fromEntries(
+      environmentVariables.filter(({ value }) => !value).map(({ name, value }) => [name, value])
+    );
+    if (Object.keys(secretEnvironmentVariables).length > 0) {
+      Log.warn(
+        `The following environment variables are secret and cannot be read locally and used when generating update bundles: ${Object.keys(
+          secretEnvironmentVariables
+        ).join(', ')}.`
+      );
+    }
+    Log.log(
+      `Loaded environment variables for ${environment} environment: ${Object.keys(
+        readableEnvironmentVariables
+      ).join(', ')}.`
+    );
+    Log.newLine();
+    process.env = {
+      ...process.env,
+      ...readableEnvironmentVariables,
     };
   }
 }
+
+async function resolveExpoConfigWithEasEnvironmentVariablesLoadedAsync({
+  environment,
+  graphqlClient,
+  getDynamicPublicProjectConfigAsync,
+}: {
+  environment: EnvironmentVariableEnvironment;
+  graphqlClient: ExpoGraphqlClient;
+  getDynamicPublicProjectConfigAsync: (options?: ExpoConfigOptions) => Promise<{
+    projectId: string;
+    exp: ExpoConfig;
+    projectDir: string;
+  }>;
+}): Promise<{
+  projectId: string;
+  exp: ExpoConfig;
+  projectDir: string;
+}> {
+  const { projectId } = await getDynamicPublicProjectConfigAsync();
+  await maybeUpdateProcessEnvWithEasEnvironmentVariablesAsync({
```

Comment:
should this revert the process.env back after executing the inner function? (the way we do in `getExpoConfigInternal`)